testing/dart/image_filter_test.dart

```diff
@@ -297,4 +298,85 @@ void main() {
       ),
     );
   });
+
+  // Tests that FilterQuality.<value> produces the expected golden file.
+  group('ImageFilter|FilterQuality', () async {
+    final ImageComparer comparer = await ImageComparer.create();
+
+    /// Draw a 100x100 red-green checkerboard pattern with 1x1 squares.
+    Future<Image> drawCheckerboard() async {
+      final PictureRecorder recorder = PictureRecorder();
+      final Canvas canvas = Canvas(recorder);
+      final Paint paint = Paint();
+      final Path pathRed = Path();
+      final Path pathGreen = Path();
+      for (int y = 0; y < 50; y++) {
+        for (int x = 0; x < 50; x++) {
+          if ((x + y).isEven) {
+            pathRed.addRect(Rect.fromLTWH(x.toDouble(), y.toDouble(), 1.0, 1.0));
+          } else {
+            pathGreen.addRect(Rect.fromLTWH(x.toDouble(), y.toDouble(), 1.0, 1.0));
+          }
+        }
+      }
+      canvas.drawPath(pathRed, paint..color = red);
+      canvas.drawPath(pathGreen, paint..color = green);
+      final Picture picture = recorder.endRecording();
+      return picture.toImage(50, 50);
+    }
+
+    /// Return the [image] magnified by a factor of 0.5 and then scaled up 10x.
+    Future<Image> shrinkAndScaleImage(Image image, FilterQuality quality) async {
+      Future<Image> shrink() async {
+        final Paint paint = Paint()..filterQuality = quality;
+        final PictureRecorder recorder = PictureRecorder();
+        final Canvas canvas = Canvas(recorder);
+        canvas.drawImage(image, Offset.zero, paint);
+
+        final Picture picture = recorder.endRecording();
+        return picture.toImage(25, 25);
+      }
+
+      Future<Image> scale(Image image) async {
+        final Paint paint = Paint()..filterQuality = quality;
+        final PictureRecorder recorder = PictureRecorder();
+        final Canvas canvas = Canvas(recorder);
+        canvas.drawImageRect(
+          image,
+          const Rect.fromLTWH(0, 0, 25, 25),
+          const Rect.fromLTWH(0, 0, 250, 250),
+          paint,
+        );
+
+        final Picture picture = recorder.endRecording();
+        return picture.toImage(250, 250);
+      }
+
+      return scale(await shrink());
+    }
+
+    test('FilterQuality.low', () async {
```

Comment:
What will the name be for these tests when sent to skia gold?  It seems like we'd want a prefix to them like we do for the impeller ones to distinguish them from web tests or impeller tests?

Reply:
Oh i checked the golden change notification.  I think we should prefix these somehow for sure.  Maybe something like `dart_ui_`?